sn_registers/src/permissions.rs

```diff
@@ -11,17 +11,17 @@ use super::Action;
 use serde::{Deserialize, Serialize};
 use std::{collections::BTreeMap, hash::Hash};
 
-/// Set of public permissions for a user.
+/// The rights of an user
 #[derive(Copy, Clone, Serialize, Deserialize, PartialEq, PartialOrd, Ord, Eq, Hash, Debug)]
-pub struct Permissions {
+pub struct UserRights {
     /// `Some(true)` if the user can write.
-    /// `Some(false)` explicitly denies this permission (even if `Anyone` has permissions).
-    /// Use permissions for `Anyone` if `None`.
+    /// `Some(false)` explicitly denies writes (even if `Anyone` has the right to write).
+    /// `None` use default: the rights for `Anyone`
     write: Option<bool>,
 }
 
-impl Permissions {
-    /// Constructs a new public permission set.
+impl UserRights {
```

Comment:
This gave me pause... `UserPermissions`? 

I guess you were trying to clarify the overall permission map vs per user perms? 

If so it feels like  `UserPermissions` would address this without adding another term into the mix?

Reply:
Make sense yeah